src/common/src/util/epoch.rs

```diff
@@ -150,20 +156,43 @@ impl EpochPair {
         Self { curr, prev }
     }
 
-    pub fn inc(&mut self) {
-        self.curr += 1;
-        self.prev += 1;
+    pub fn inc_for_test(&mut self) {
+        self.prev = self.curr;
+        self.curr += EPOCH_INC_MIN_STEP_FOR_TEST;
     }
 
-    pub fn inc_for_test(&mut self, inc_by: u64) {
-        self.prev = self.curr;
+    pub fn new_test_epoch(curr: u64) -> Self {
+        if !is_max_epoch(curr) {
+            assert!(curr >= EPOCH_INC_MIN_STEP_FOR_TEST);
+            assert!((curr & EPOCH_SPILL_TIME_MASK) == 0);
+        }
+        Self::new(curr, curr - EPOCH_INC_MIN_STEP_FOR_TEST)
+    }
+}
+/// As most unit tests initializ a new epoch from a random value (e.g. 1, 2, 233 etc.), but the correct epoch in the system is a u64 with the last `EPOCH_AVAILABLE_BITS` bits set to 0.
+/// This method is to turn a a random epoch into a well shifted value.
+pub const fn test_epoch(value: u64) -> u64 {
+    value << EPOCH_AVAILABLE_BITS
+}
 
-        self.curr += inc_by;
+/// There are numerous operations in our system's unit tests that involve incrementing or decrementing the epoch.
+/// These extensions for u64 type are specifically used within the unit tests.
+#[ext(EpochExt)]
+pub impl u64 {
+    fn inc_epoch(&mut self) {
+        *self += EPOCH_INC_MIN_STEP_FOR_TEST;
     }
 
-    pub fn new_test_epoch(curr: u64) -> Self {
-        assert!(curr > 0);
-        Self::new(curr, curr - 1)
+    fn dec_epoch(&mut self) {
+        *self -= EPOCH_INC_MIN_STEP_FOR_TEST;
+    }
+
+    fn next_epoch(self) -> u64 {
+        self + EPOCH_INC_MIN_STEP_FOR_TEST
+    }
+
+    fn prev_epoch(self) -> u64 {
+        self - EPOCH_INC_MIN_STEP_FOR_TEST
```

Comment:
It seems that extending `u64` cannot prevent future miss use. It's very likely that someone will write `let epoch = 1u64` and `epoch += 1` later sometime.

What about:

```rust
#[derive(Clone, Copy)]
pub struct TestEpoch(u64);

impl TestEpoch {
  pub fn new(val: u64) -> Self {
    Self(val << EPOCH_AVAILABLE_BITS);
  }

  pub fn inc(&mut self) {
    // ...
  }

  pub fn next(self) -> Self {
    // ...
  }

  // ...
}
```

And change the argument data type of  `Barrier::new_test_barrier` and `MessageSender::push_barrier` (and maybe some other functions) to `TestEpoch`?

Reply:
I think it's fine, because the invalid epoch will fail the unit test. To pass the unit test, we will have to change to increment the epoch in a correct way. The extended trait is not for avoiding misuse, but only some util methods to help using the correct epoch.